src/mixer/generate_audio.py

```diff
@@ -44,18 +44,35 @@ def load_voice_memos() -> List[AudioSegment]:
         raise NoVoiceMemosFoundError(f"No voice memos found in {VOICE_DIR}")
 
     voice_segs: List[AudioSegment] = []
+    cumulative_position_ms = INTRO_MS
+
     for f in voice_files:
-        logger.info(f"Loading {f.name}")
+        logger.info(f"Loading voice memo: {f.name}")
         segment = AudioSegment.from_file(str(f))
+        duration_ms = len(segment)
+
         if len(segment) > MAX_LENGTH_MS:
-            logger.info(f"Voice memo {f.name} exceeds 3m 10s limit, truncating...")
+            logger.info(
+                f"Voice memo {f.name} exceeds 3m 10s limit, "
+                f"truncating from {duration_ms}ms to {MAX_LENGTH_MS}ms"
+            )
             segment = segment[:MAX_LENGTH_MS]
+            duration_ms = MAX_LENGTH_MS
 
         normalized_seg = (
             normalize(segment).fade_in(VOICE_FADE_MS).fade_out(VOICE_FADE_MS)
         )
         voice_segs.append(normalized_seg)
 
+        timestamp = datetime.datetime.now().isoformat()
+        logger.info(
+            f"[{timestamp}] Voice memo loaded: {f.name} | Duration: {duration_ms}ms | "
```

Comment:
DevinAI: our logging handler includes timestamps, we can remove all the extra timestamps